Add single-threaded config to Filament.

prideout opened the pull request:
This adds a tick method to Engine and disables a couple components
in Renderer (FrameSkipper and FrameInfoManager).

This will make it easier to support WebGL, and will allow us to remove
some of the command buffer debugging stuff that we added for Vulkan.

romainguy commented on filament/src/Engine.cpp:
```diff
@@ -699,6 +707,18 @@ void* FEngine::streamAlloc(size_t size, size_t alignment) noexcept {
     return getDriverApi().allocate(size, alignment);
 }
 
+void FEngine::tick() noexcept {
```
How about `execute()` or `run()` or something? I would expect `tick()` to be time- or physics-simulation related.

prideout replied:
I think I'll go with `execute` since I've often seen `run` often used to mean "start the loop".

romainguy commented on filament/include/filament/Engine.h:
```diff
@@ -335,6 +335,14 @@ class UTILS_PUBLIC Engine {
+     * Note that the OpenGL context that was created during Engine construction needs to be
+     * current at the time that this is called.
+     */
+    void tick() noexcept;
```
The comment is kind of meaningless in a public header since we build the GL context and we make it current. Also we don't necessarily have a GL backend :)

romainguy commented on filament/src/Renderer.cpp:
```diff
@@ -302,9 +314,9 @@ void FRenderer::endFrame() {
-#if EXTRA_TIMING_INFO
-    if (UTILS_UNLIKELY(frameInfoManager.isLapRecordsEnabled())) {
-        auto history = frameInfoManager.getHistory();
+#if EXTRA_TIMING_INFO and UTILS_HAS_THREADING
```
☝️

romainguy commented:
Looks like this change breaks the Linux build?

prideout commented on filament/src/Engine.cpp:
```diff
@@ -79,6 +79,14 @@ FEngine* FEngine::create(Backend backend, ExternalContext* externalContext, void
     // initialize all fields that need an instance of FEngine
     // (this cannot be done safely in the ctor)
 
+    if (!UTILS_HAS_THREADING) {
```
Done.

prideout commented on filament/include/filament/Engine.h:
```diff
@@ -335,6 +335,14 @@ class UTILS_PUBLIC Engine {
     }
 
+   /**
+     * Invokes one iteration of the render loop, used only on single-threaded platforms.
```
Done.

prideout commented:
Linux build is passing and I addressed all the feedback from @pixelflinger.  The loop now calls `execute` internally, but please note that the public `execute` does a flush before calling the internal `execute`.
